### PR TITLE
Added fix for specifying every option for reporting using 'a'

### LINKED ISSUE #960: Add shortcut to report all possible cases with `-r` option
Currently the option for `-r` reads:

```
  -r chars              show extra test summary info as specified by chars
                        (f)ailed, (E)error, (s)skipped, (x)failed, (X)passed
                        (w)warnings.
```

So to obtain all possible reports the user has to use `-rfEsxXw`.

It would be handy to add a shortcut (for example `-rA`, for "all") that has the same effect of explicitly passing all options.

(Originally from #527)

### PR DESCRIPTION
**Pull request by Akasurde**
Fixes #960 

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

### CODE REVIEW

**Comment by The-Compiler** [comment]
Could you add a test for this? You can probably just parametrize `test_reportchars` and `test_reportchars_error` in `testing/test_skipping.py`.

File: _pytest/terminal.py [hunk 1]
```diff
@@ -64,8 +64,10 @@ def getreportopt(config):
     reportchars = config.option.reportchars
     if reportchars:
         for char in reportchars:
-            if char not in reportopts:
+            if char not in reportopts and char != 'a':
```
Comment (The-Compiler) [inline _pytest/terminal.py]: The code above uses `"` for strings, so it'd be nice if you could do the same everywhere for consistency.

Comment (Akasurde) [inline _pytest/terminal.py]: OK.

Comment (Akasurde) [inline _pytest/terminal.py]: Also, I will add test case for this.

**Comment by nicoddemus** [comment]
Thanks for the PR @Akasurde! :smile:

Besides the test as @The-Compiler mentioned, could you please also:

* Add yourself to `AUTHORS` and a note to the `CHANGELOG`?
* Mention this in the help for [pytest/terminal](https://github.com/pytest-dev/pytest/blob/master/_pytest/terminal.py#L23)

**Comment by nicoddemus** [comment]
Merged, thanks @Akasurde!

**Pull request merged**